### PR TITLE
Comments new ui

### PR DESCRIPTION
**Pull request by vidya-ram**
Dependency https://github.com/hasgeek/baseframe/pull/326

### CODE REVIEW

File: funnel/templates/macros.html.jinja2 [hunk 1]
```diff
@@ -760,84 +760,141 @@
 {% macro comments_tree() %}
   {% raw %}
     <div id="comments-wrapper" v-cloak>
-      <div class="ajax-form">
-        <component :is="Form"></component>
-        <button class="mui-btn mui-btn--raised mui-btn--accent ajax-form_cancelbtn" @click="closeForm($event)" v-if="commentForm">Cancel</button>
-        <p class="mui--text-subhead mui--text-danger mui--text-right" v-if="errorMsg">{{ errorMsg }}</p>
-      </div>
+      <a class="mui-btn mui-btn--small mui-btn--raised mui-btn--primary" :href="loginUrl" data-cy="login-btn" v-if="!isuserloggedin">Login to leave a comment</a>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:763]: `{% trans %}` needed.

File: funnel/templates/macros.html.jinja2 [hunk 2]
```diff
@@ -760,84 +760,141 @@
...
       <p class="mui-panel mui--bg-accent" v-if="isuserloggedin && !isuserparticipant">You need to be a participant to comment.</p>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:764]: `{% trans %}` needed.

File: funnel/templates/macros.html.jinja2 [hunk 3]
```diff
@@ -760,84 +760,141 @@
...
-      <a class="post-form-block" @click="fetchForm($event, newCommentUrl)" aria-label="Post comment" data-cy="post-comment" v-if="isuserparticipant && !commentForm">
-        <div class="user">
+      <div class="mui--hidden-md mui--hidden-lg mui--hidden-xl" v-if="showmodal">
+        <div class="mobile-nav mui--z1">
+          <span aria-label="Close" class="mui--text-dark mobile-nav__icon" data-action="Close" @click="closeNewCommentForm($event)"><faicon :icon="'times'" :icon_size="'subhead'"></faicon></span><span class="mui--text-dark mobile-nav__headline">{{ formTitle }}</span>
+        </div>
+      </div>
+      <a href="javascript:void(0)" class="post-form-block nounderline js-new-comment-form" :class="[commentForm ? 'mui--hide' : '']" @click="showNewCommentForm($event, 'comment_message')" aria-label="Post comment" data-cy="post-comment" v-if="isuserparticipant" role="button">
+        <span class="post-form mui--hidden-xs mui--hidden-sm">Post a comment...</span>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:771]: `{% trans %}`

File: funnel/templates/macros.html.jinja2 [hunk 4]
```diff
@@ -760,84 +760,141 @@
...
+        <div class="user mui--hidden-xs mui--hidden-sm">
           <div class="user__box">
             <useravatar :user='user' :addprofilelink=false></useravatar>
-            <input type="text" placeholder="Post comment…" />
           </div>
         </div>
+        <span class="comment-new-mobile mui--text-hyperlink mui--hidden-md mui--hidden-lg mui--hidden-xl"><faicon :icon="'comment-alt'" :icon_size="'subhead'" :css_class="'fa-icon--margin'"></faicon>New comment</span>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:777]: `{% trans %}`

File: funnel/templates/macros.html.jinja2 [hunk 5]
```diff
@@ -760,84 +760,141 @@
...
+        <div class="mui--hidden-md mui--hidden-lg mui--hidden-xl" v-if="formOpened">
+          <div class="mobile-nav mui--z1">
+            <span aria-label="Close" class="mui--text-dark mobile-nav__icon" data-action="Close" @click="closeAllForms($event)"><faicon :icon="'times'" :icon_size="'subhead'"></faicon></span><span class="mui--text-dark mobile-nav__headline">{{ commentFormTitle }}</span>
+          </div>
+        </div>
+        <div class="comment__header">
+          <button class="mui-btn mui-btn--nostyle" aria-label="Hide comment" aria-expanded="true" v-if="!hide" @click="collapse(true)">
+            <faicon :icon="'caret-circle-up'" :icon_size="'subhead'"></faicon>
+          </button>
+          <button class="mui-btn mui-btn--nostyle" aria-expanded="false" aria-label="Collapse comment" v-else @click="collapse(false)">
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:808]: ARIA attrs also need `{% trans %}`

File: funnel/templates/macros.html.jinja2 [hunk 6]
```diff
@@ -760,84 +760,141 @@
...
+          </button>
+          <div class="comment__header__details user">
+            <div class="user__box" v-if="!comment.state">
+              <useravatar :user='comment.user' :addprofilelink=true></useravatar>
+              <div class="user__box__header comment__header__details__user">
+                <h3 class="mui--text-body2 user__box__fullname zero-bottom-margin">
+                  <span class="commenter">{{ comment.user.fullname }}</span>
+                  <span class="badge" v-for="badge in comment.badges">{{ badge }}</span>
+                  <span class="commenter mui--text-light mui--text-caption" v-if="comment.state && comment.state.DELETED">[deleted]</span>
+                  <span class="commenter mui--text-light mui--text-caption" v-if="comment.state && comment.state.SPAM">[removed]</span>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:819]: Both need `{% trans %}`

File: funnel/templates/macros.html.jinja2 [hunk 7]
```diff
@@ -760,84 +760,141 @@
...
-                <a class="mui--text-caption mui--text-light comment--permalink" :href="'#c-' + comment.uuid_b58">Posted {{ created_at_age }}</a>
-                <span class="mui--text-caption mui--text-light" v-if="comment.edited_at">(edited {{ edited_at_age }})</span>
-              </div>
-              <div>
-                <a class="link-icon mui--text-light link comment__body__links" data-cy="reply" v-if="isuserparticipant"  @click="fetchForm($event, comment.urls.reply)"><faicon :icon="'comment-alt'" :baseline=true></faicon><span class="mui--hidden-xs mui--hidden-sm">Reply</span></a>
-                <a class="link-icon mui--text-light link comment__body__links" data-cy="report-spam" v-if="isuserparticipant || iscommentmoderator" @click="fetchForm($event, comment.urls.report_spam)"><faicon :icon="'flag-alt'":baseline=true></faicon><span class="mui--hidden-xs mui--hidden-sm">Report</span></a>
+              <a class="mui--text-caption mui--text-light comment--permalink hg-link-btn" :href="'#c-' + comment.uuid_b58">Posted {{ created_at_age }}</a>
+              <span class="mui--text-caption mui--text-light" v-if="comment.edited_at">(edited {{ edited_at_age }})</span>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:831]: Trans

File: funnel/templates/macros.html.jinja2 [hunk 8]
```diff
@@ -760,84 +760,141 @@
...
+              <div class="js-comment-form">
+                <a href="javascript:void(0)" class="link-icon mui--text-light link comment__body__links" data-cy="reply" v-if="isuserparticipant"  @click="activateSubForm($event, COMMENTACTIONS.REPLY, 'reply-comment-textarea-' + comment.uuid_b58)"><faicon :icon="'comment-alt'" :baseline=true></faicon><span class="mui--hidden-xs mui--hidden-sm">Reply</span></a>
+                <a href="javascript:void(0)" class="link-icon mui--text-light link comment__body__links" data-cy="report-spam" v-if="isuserparticipant || iscommentmoderator" @click="activateSubForm($event, COMMENTACTIONS.REPORTSPAM)"><faicon :icon="'flag-alt'":baseline=true></faicon><span class="mui--hidden-xs mui--hidden-sm">Report</span></a>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:834]: Trans

File: funnel/templates/macros.html.jinja2 [hunk 9]
```diff
@@ -760,84 +760,141 @@
...
                 <div class="mui-dropdown">
                   <a href="javascript:void(0)" class="link-icon project-links mui--text-light link comment__body__links" data-mui-toggle="dropdown" data-action="Share dropdown" data-cy="share-project"><faicon :icon="'share-alt-square'" :baseline=true></faicon><span class="mui--hidden-xs mui--hidden-sm">Share</span></a>
                   <sharedropdown :url='comment.urls.view' :title='comment.title'></sharedropdown>
                 </div>
-                <a href="javascript:void(0)" class="link-icon mui--text-light hg-link-btn mui--hide right-padding" :data-title="comment.title" :data-url="comment.absolute_url + '&amp;utm_campaign=webshare'"><faicon :icon="'share-alt'" :baseline=true></faicon></a>
+                <a href="javascript:void(0)" class="link-icon mui--text-light hg-link-btn  mui--hide right-padding" :data-title="comment.title" :data-url="comment.absolute_url + '&amp;utm_campaign=webshare'" :data-text=comment.message.text><faicon :icon="'share-alt'" :baseline=true></faicon></a>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:839]: Extra space

File: funnel/templates/macros.html.jinja2 [hunk 10]
```diff
@@ -760,84 +760,141 @@
...
-                    <li><a class="mui--text-body2 comment__body__links" data-cy="delete" v-if="comment.urls.delete"  @click="fetchForm($event, comment.urls.delete)"><faicon :icon="'trash-alt'" :icon_size="'caption'" :baseline=true :css_class="'mui--text-light fa-icon--margin'"></faicon> Delete</a></li>
+                    <li><a href="javascript:void(0)" class="mui--text-body2 comment__body__links" data-cy="edit" v-if="comment.urls.edit" @click="activateSubForm($event, COMMENTACTIONS.EDIT, 'edit-comment-textarea-' + comment.uuid_b58)"><faicon :icon="'edit'" :icon_size="'caption'" :baseline=true :css_class="'mui--text-light fa-icon--margin'" ></faicon> Edit</a></li>
+                    <li><a href="javascript:void(0)" class="mui--text-body2 comment__body__links" data-cy="delete" v-if="comment.urls.delete"  @click="activateSubForm($event, COMMENTACTIONS.DELETE)"><faicon :icon="'trash-alt'" :icon_size="'caption'" :baseline=true :css_class="'mui--text-light fa-icon--margin'"></faicon> Delete</a></li>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:844]: Trans

File: funnel/templates/macros.html.jinja2 [hunk 11]
```diff
@@ -760,84 +760,141 @@
...
+                      <div class="user__box">
+                        <useravatar :user='user' :addprofilelink=false></useravatar>
+                      </div>
+                    </div>
+                  </form>
+                  <p class="mui--text-subhead mui--text-danger mui--text-right" v-if="errorMsg">{{ errorMsg }}</p>
+                </div>
+                <div class="ajax-form ajax-form--block" :class="[formOpened ? 'ajax-form--mob' : '']" v-if="deleteForm">
+                  <form method="POST" accept-charset="UTF-8" class="mui-form" @submit.prevent="submitCommentForm('delete-comment-form-' + comment.uuid_b58, comment.urls.delete, 2)" :id="'delete-comment-form-' + comment.uuid_b58" data-cy="delete-form">
+                    <p class="ajax-form__title">Delete this comment?</p>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:877]: Trans

File: funnel/templates/macros.html.jinja2 [hunk 12]
```diff
@@ -760,84 +760,141 @@
...
+                    <button type="submit" class="mui-btn mui-btn--small mui-btn--raised mui-btn--primary">Confirm</button>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:878]: Trans

File: funnel/templates/macros.html.jinja2 [hunk 13]
```diff
@@ -760,84 +760,141 @@
...
+                    <button class="mui-btn mui-btn--small mui-btn--raised mui-btn--accent ajax-form__cancelbtn" @click="closeAllForms($event, 'deleteForm')">Cancel</button>
+                  </form>
+                  <p class="mui--text-subhead mui--text-danger mui--text-right" v-if="errorMsg">{{ errorMsg }}</p>
+                </div>
+                <div class="ajax-form ajax-form--block" :class="[formOpened ? 'ajax-form--mob' : '']" v-if="spamForm">
+                  <form method="POST" accept-charset="UTF-8" class="mui-form" @submit.prevent="submitCommentForm('spam-comment-form-' + comment.uuid_b58, comment.urls.report_spam, 3)" :id="'spam-comment-form-' + comment.uuid_b58">
+                    <p class="ajax-form__title">Do you want to mark this comment as spam?</p>
+                    <button type="submit" class="mui-btn mui-btn--small mui-btn--raised mui-btn--primary">Confirm</button>
+                    <button class="mui-btn mui-btn--small mui-btn--raised mui-btn--accent ajax-form__cancelbtn" @click="closeAllForms($event, 'spamForm')">Cancel</button>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:887]: Trans

**Review by jace — CHANGES_REQUESTED** [review]
Several `{% trans %}` tags are missing.

File: funnel/templates/layout.html.jinja2 [hunk 14]
```diff
@@ -54,7 +54,6 @@
   <meta name="apple-mobile-web-app-capable" content="yes" />
   <meta name="apple-mobile-web-app-status-bar-style" content="#e3e1e1" />
   <meta name="apple-mobile-web-app-title" content="js" />
-  <link rel="icon" sizes="192x192" href="{{ url_for('static', filename='img/android-chrome-192x192.png', v=2, _external=true) }}" />
```
[the remark this answers is not in the document]
Comment (vidya-ram) [reply funnel/templates/layout.html.jinja2:57]: This is duplicate and wrong link

**Pull request merged**